Migration : Frontend

Nazi-pikachu opened the pull request:
## Description
To migrate the frontend from previous seperate repo to the main repo

## Changes Made
Generated a react-app using nx

## Related Issues
- [x] #76 
- [ ]  #52 

## Testing Instructions

1. Follow the setup readme 
2. run `nx serve frontend`

<!--
## Checklist
- [ ] Reviewed code for quality and adherence to coding standards
- [ ] Added/updated relevant tests for the changes made
- [ ] Updated documentation, if applicable
- [ ] Verified that the build passes
- [ ] Ensured there are no linting errors  -->

## Screenshots (if applicable)
Current output 
![image](https://github.com/Samagra-Development/yaus/assets/63303757/da44db8f-9fe2-44bf-91a0-868f94fea925)

## Additional Notes
Its an intial PR in which the frontend code has been migrated to frontend app follow test instructions to know how to run the frontend.

yuvrajsab commented on apps/frontend/src/app/Protect.js:
```diff
@@ -0,0 +1,21 @@
+import { useNavigate } from "react-router-dom";
+
+function Protect(props) {
+  let Cmp = props.Cmp;
```
use proper names for variables.

yuvrajsab commented on apps/frontend/src/app/apis/index.js:
```diff
@@ -0,0 +1,100 @@
+  email: "test@gmail.com",
+  password: "test@123",
+};
+const baseUrl = "https://fa.chakshu-rd.samagra.io/api";
```
access these from an env or app constants.

Shruti3004 replied:
@Nazi-pikachu

yuvrajsab commented on apps/frontend/src/app/apis/index.js:
```diff
@@ -0,0 +1,100 @@
+};
+const baseUrl = "https://fa.chakshu-rd.samagra.io/api";
+const getSignUpReqBody = (user) => {
+  console.log(
+    JSON.stringify({
+      registration: {
+        applicationId: "650c50b5-eae9-442e-88de-fc36be10a21b", // TO DO : Change this to env variable
+      },
+      user: {
+        name: "test",
+        email: "test@gmail.com",
+        password: "test@123",
+      },
+    })
+  );
```
remove unnecessary logs

yuvrajsab commented on apps/frontend/src/app/assets/styles/responsive.css:
```diff
@@ -0,0 +1,73 @@
+    margin: 0 7px;
+  }
+  .layout-dashboard .header-control .btn-sign-in span {
+    display: inline;
+  }
+  .profile-nav-bg {
+    margin-top: -87.8px;
+  }
+  .card-profile-head {
+    margin: -53px 24px 24px;
+  }
+}
+
+@media (min-width: 992px) {
+  .card-billing-info.ant-card .ant-card-body {
+    display: flex;
+  }
+  .layout-dashboard-rtl {
+    overflow: auto;
+  }
+  .layout-dashboard-rtl .ant-layout-sider.sider-primary {
+    margin: 20px 20px 0 0;
+    height: calc(100vh - 20px);
+  }
+  .layout-dashboard-rtl .ant-layout-sider.sider-primary {
+    right: 0;
+    left: auto;
+  }
+  .layout-dashboard-rtl .ant-layout-sider.sider-primary {
+    right: 0;
+    left: auto;
+  }
+  .layout-dashboard-rtl .ant-layout {
+    margin-right: 270px;
+    margin-left: 0;
+  }
+  .layout-dashboard-rtl .ant-layout-footer {
+    margin: 0 20px 20px 0;
+  }
+}
```
min widths should come in ascending order. e.x 768 then 992 then 1024.

yuvrajsab commented on apps/frontend/src/app/assets/styles/responsive.css:
```diff
@@ -0,0 +1,73 @@
+    margin-top: -87.8px;
+  }
+  .card-profile-head {
+    margin: -53px 24px 24px;
```
negative margins should be avoided.

yuvrajsab commented on apps/frontend/src/app/pages/FormHomePage.js:
```diff
@@ -0,0 +1,110 @@
+import { nanoid } from "nanoid";
+
+function FormHomePage() {
+  const [state, setState] = useState({
+    userID: "0fe6ff38-fc46-11ec-b939-0242ac120001",
+    url: "",
+    project: "0fe6ff38-fc46-11ec-b939-0242ac120002",
+    customHashId: nanoid(6),
+    titleImage: "",
+    urlImg: "",
+    description: "",
```
should not be hardcoded like that.

yuvrajsab commented on apps/frontend/src/app/pages/FormTagsPage.js:
```diff
@@ -0,0 +1,251 @@
+                    count={12}
+                    style={{ paddingTop: "10px" }}
+                  />
+                  &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
```
find another way to add spaces. remove this &nbsp;

Nazi-pikachu commented:
Progress on this PR Till now

- Frontend Migrated to monorepo
- Migrated to react-router-dom v6
- Fixed few pages for responsiveness
- Refactored the entire Frontend as per the standards
- Deployed to vercel to ease tracking

Shruti3004 commented on apps/frontend/src/app/Protect.js:
```diff
@@ -0,0 +1,21 @@
+import React, { useEffect } from "react";
```
@Nazi-pikachu can we move this in folder name `utils`

Nazi-pikachu replied:
Resolved

Shruti3004 commented on apps/frontend/src/app/app.jsx:
```diff
@@ -0,0 +1,17 @@
+  return (
+    <div className="App">
+      <Routes>
+        <Route path="/sign-up" element={<SignUp />} />
```
@Nazi-pikachu please create the constants folder for all the routes

Nazi-pikachu replied:
resolved

Shruti3004 commented on apps/frontend/src/app/app.module.css:
```diff
@@ -0,0 +1 @@
+/* Your styles goes here. */
```
Delete this file

Nazi-pikachu replied:
resolved

Shruti3004 commented on apps/frontend/src/app/assets/constants/Config/DashboardConfig/index.js:
```diff
@@ -0,0 +1,30 @@
+import { piedata } from "app/assets/constants/mockData";
```
@Nazi-pikachu please take the constants and styles folder out of it

Nazi-pikachu replied:
resolved
